canvas_modules/common-canvas/src/object-model/object-model.js

```diff
@@ -947,6 +947,13 @@ export default class ObjectModel {
 		return notificationMessages;
 	}
 
+	deleteNotificationMessages(ids) {
+		// accept a single key or array of keys
+		const filterIds = Array.isArray(ids) ? ids : [ids];
+		const newMessages = this.getNotificationMessages().filter((message) => !filterIds.includes(message.id));
+		this.store.dispatch({ type: "SET_NOTIFICATION_MESSAGES", data: newMessages });
+	}
```

Comment:
It threw me for a minute when I saw the variables was called 'newMessages'.  Maybe 'retainedMessages' would be better.
I think I might have been tempted to add a new redux action called DELETE_NOTIFICATION_MESSAGES and do the work inside the body of the redux action - but I guess this will work.

Reply:
I see now the phrasing is a bit confusing. "newMessages" made sense to me since this new array of messages comprises the new entirety of the message log, though it is just a filtered, smaller version of the previous state.

This redux method/naming is also used for clearing all messages (either through the sidepanel api in the test harness or the new clear all button, so I thought it'd be more consistent this way.